src/xdealer_channel.cpp

```diff
@@ -0,0 +1,81 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include "xdearler_channel.hpp"
+
+namespace xeus
+{
+
+    xdearler_channel::xdearler_channel(zmq::context_t& context,
+                                 const std::string& user_name,
+                                 const xeus::xconfiguration& config)
+        : m_shell(context, zmq::socket_type::dealer)
+        , m_control(context, zmq::socket_type::dealer)
```

Comment:
So actually, `xdealer_channel` should handle a single channel, and `xclient_zmq_impl` should have two `xdealer_channel` members, one for the shell and one for the control.